types/src/transaction/authenticator.rs

```diff
@@ -108,6 +133,25 @@ impl TransactionAuthenticator {
                 public_key,
                 signature,
             } => signature.verify(raw_txn, public_key),
+            Self::FeePayer {
+                sender,
+                secondary_signer_addresses,
+                secondary_signers,
+                fee_payer_address,
+                fee_payer_signer,
+            } => {
+                let message = RawTransactionWithData::new_fee_payer(
```

Comment:
Can you leave a comment here that this is similar to multiagent transactions and is necessary so that the fee payer portion cannot be "stripped" out of a transaction that has been signed by senders?